tests/kernel/gen_isr_table/src/main.c

```diff
@@ -107,11 +108,16 @@ void isr4(void *param)
  * accesses to trigger_check around calls to trigger_irq.
  */
 __attribute__((optimize("-O0")))
```

Comment:
just afraid this may hide some issues. usually we can add instruction barrier  to the suspected code segment